ql/src/java/org/apache/hadoop/hive/ql/metadata/HiveStorageHandler.java

```diff
@@ -304,6 +305,22 @@ default boolean canComputeQueryUsingStats(org.apache.hadoop.hive.ql.metadata.Tab
   default StorageFormatDescriptor getStorageFormatDescriptor(Table table) throws SemanticException {
     return null;
   }
+
+  /**
+   * Checks whether the table supports appending data files to the table.
+   * @param table the table
+   * @return true if the table can append files directly to the table
+   * @throws SemanticException in case of any error.
+   */
+  default boolean supportsAppendData(Table table) throws SemanticException {
+    throw new SemanticException(ErrorMsg.LOAD_INTO_NON_NATIVE.getMsg());
```

Comment:
should we return `false` here otherwise i don't see why we need this method?

Reply:
done, thnx